great_expectations/core/datasource_dict.py

```diff
@@ -17,11 +17,27 @@
         AbstractDataContext,
     )
     from great_expectations.data_context.store.datasource_store import DatasourceStore
+    from great_expectations.datasource.fluent.interfaces import DataAsset
     from great_expectations.datasource.new_datasource import BaseDatasource
 
 logger = logging.getLogger(__name__)
 
 
+@runtime_checkable
+class SupportsInMemoryDataAssets(Protocol):
+    @property
+    def assets(self) -> list[DataAsset]:
+        ...
+
+    def add_dataframe_asset(self, **kwargs) -> DataAsset:
+        ...
+
+
+@runtime_checkable
+class InMemoryDataAsset(Protocol):
+    type: Literal["dataframe"] = "dataframe"
```

Comment:
```suggestion
    @property
    def type(self) -> Literal["dataframe"]:
       ...
```

Reply:
```
class InMemoryDataAsset(Protocol):
    type: Literal["dataframe"] = "dataframe"
```

The above is saying that this is a class attribute, but it's an instance attr.

I would suggest changing it but it's not urgent.